contrib-compiler.md

```diff
@@ -0,0 +1,99 @@
+
+
+Thanks for interest in contributions.
+
+Note currently our philosphy is to make release build as easy as possible, while dev build is fairly complex at this time.
+
+## Set up dev dependencies
+
+- Having [https://opam.ocaml.org/](opam) installed
+
+    opam switch 4.02.3+buckle-master # use our OCaml compiler
+    opam install camlp4 cppo  <1>
+
+   Both `Camlp4` and `cppo` are used to generate OCaml code for pre-processing. 
+   Switch to our compiler makes your life easier (avoid confusion with diferent compiler versions)
+- Having [https://nodejs.org/](NodeJS) installed
+- Having GNU Make installed
+- OS: Mac/Linux (Note BuckleScript works well on Windows, but the dev mode is not tested)
+- Our dev build depends on a submodule
+
+  ```sh
+  git submodule --update
+  ``` 
+
+All source code are in `jscomp` directory, below assume that you are working in `jscomp` directory.
+
```

Comment:
In my opinion, it would be most important for contributors to know how to make changes to the compiler, build it, and run the modified version.

What if you were to add the short version In a section here called something like "development workflow":

in the jscomp directory:
- Make changes to source files in any of these directories _____. Do not edit the files in __, as they are generated using bspack. See below for more details.
- Run ./build.sh to build the compiler. See below for more details on build.sh.
- To run the compiler on OCaml or Bucklescript source files ___ *I'm not sure about this, and couldn't get it working. I'd expect the directions to be something like:*

> Anywhere in this repo, do `npm link` to symlink your global bsb-platform to the build you just created. Then you can use your development build as described in the the [manual](http://bloomberg.github.io/bucklescript/Manual.html#_get_started). To undo the symlink, do `npm unlink` anywhere in this repo."